File: src/main/java/com/bajdcc/util/ResourceLoader.java
package com.bajdcc.util;

import org.apache.log4j.Logger;

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.UnsupportedEncodingException;
import java.net.URISyntaxException;
import java.util.stream.Collectors;

public class ResourceLoader {

	private static Logger logger = Logger.getLogger("loader");
	private static String sep = System.lineSeparator();

	private static final String PACKAGE_NAME = "/com/bajdcc/code";

	private static String getPrefix(String name) {
		if (name.startsWith("Module")) {
			if (name.startsWith("ModuleUser")) {
				return "/module/user/" + name;
			} else if (name.startsWith("ModuleStd")) {
				return "/module/std/" + name;
			}
			return "/module/" + name;
		} else if (name.startsWith("U")) {
			if (name.startsWith("URFile")) {
				return "/os/user/routine/file/" + name;
			} else if (name.startsWith("UR")) {
				return "/os/user/routine/" + name;
			} else if (name.startsWith("UI")) {
				return "/os/ui/" + name;
			}  else {
				return "/os/user/" + name;
			}
		} else if (name.startsWith("IR")) {
			return "/os/irq/" + name;
		} else if (name.startsWith("OS")) {
			return "/os/kern/" + name;
		} else if (name.startsWith("TK")) {
			return "/os/task/" + name;
		}  else {
			return "/os/user/" + name;
		}
	}

	public static String load(Class cls) {
		BufferedReader buffer = null;
		try {
			logger.debug("Load txt: " + cls.getResource(PACKAGE_NAME + getPrefix(cls.getSimpleName()) + ".txt").toURI());
			buffer = new BufferedReader(new InputStreamReader(cls.getResourceAsStream(PACKAGE_NAME + getPrefix(cls.getSimpleName()) + ".txt"), "UTF-8"));
			return buffer.lines().collect(Collectors.joining(System.lineSeparator()));
		} catch (UnsupportedEncodingException | URISyntaxException e) {
			e.printStackTrace();
		}
		return "";
	}
}

File: src/main/kotlin/com/bajdcc/util/ResourceLoader.kt
package com.bajdcc.util

import org.apache.log4j.Logger

import java.io.BufferedReader
import java.io.InputStreamReader
import java.io.UnsupportedEncodingException
import java.net.URISyntaxException
import java.util.stream.Collectors

object ResourceLoader {

    private val logger = Logger.getLogger("loader")
    private const val PACKAGE_NAME = "/com/bajdcc/code"

    private fun getPrefix(name: String): String =
            with(name) {
                when {
                    startsWith("Module") -> when {
                        startsWith("User", 6) -> "/module/user/$name"
                        startsWith("Std", 6) -> "/module/std/$name"
                        else -> "/module/$name"
                    }
                    else -> when {
                        startsWith("U") -> when {
                            startsWith("RFile", 1) -> "/os/user/routine/file/$name"
                            startsWith("R", 1) -> "/os/user/routine/$name"
                            startsWith("I", 1) -> "/os/ui/$name"
                            else -> "/os/user/$name"
                        }
                        startsWith("IR") -> "/os/irq/$name"
                        startsWith("OS") -> "/os/kern/$name"
                        startsWith("TK") -> "/os/task/$name"
                        else -> "/os/user/$name"
                    }
                }
            }

    fun load(cls: Class<*>): String {
        val buffer: BufferedReader
        try {
            val path = PACKAGE_NAME + getPrefix(cls.simpleName) + ".txt"
            logger.debug("Load txt: " + cls.getResource(path).toURI())
            buffer = BufferedReader(InputStreamReader(cls.getResourceAsStream(path), "UTF-8"))
            val code = buffer.lines().collect(Collectors.joining(System.lineSeparator()))
            return if (code.isEmpty()) ";" else code
        } catch (e: UnsupportedEncodingException) {
            e.printStackTrace()
        } catch (e: URISyntaxException) {
            e.printStackTrace()
        }
        return ""
    }
}
